py-denormalized/src/errors.rs

```diff
@@ -0,0 +1,75 @@
+// Code adapted from python-datafusion https://github.com/apache/datafusion-python
+
+use core::fmt;
+use std::error::Error;
+use std::fmt::Debug;
+
+use datafusion::arrow::error::ArrowError;
+use pyo3::{exceptions::PyException, PyErr};
+use denormalized::common::error::DenormalizedError as InnerDenormalizedError;
+
+pub type Result<T> = std::result::Result<T, DenormalizedError>;
+
+#[derive(Debug)]
+pub enum DenormalizedError {
+    ExecutionError(InnerDenormalizedError),
```

Comment:
does this enum have to exactly match the enum in denormalized-rs?

Reply:
`InnerDenormalizedError` is the enum from denormalized-rs while the `DenormalizedError`a defined here wraps the one defined in the rust library while also allowing us to add some python related errors. This code was adapted straight from the python-datafusion library